src/Build/BuildCheck/Checks/NoEnvironmentVariablePropertyCheck.cs

```diff
@@ -39,39 +44,53 @@ public override void Initialize(ConfigurationContext configurationContext)
         foreach (CustomConfigurationData customConfigurationData in configurationContext.CustomConfigurationData)
         {
             bool? isVerboseEnvVarOutput = GetVerboseEnvVarOutputConfig(customConfigurationData, RuleId);
-            _isVerboseEnvVarOutput = isVerboseEnvVarOutput.HasValue && isVerboseEnvVarOutput.Value;           
+            _isVerboseEnvVarOutput = isVerboseEnvVarOutput.HasValue && isVerboseEnvVarOutput.Value;
         }
+
+        CheckScopeClassifier.NotifyOnScopingReadiness += (string? projectFilePath) =>
```

Comment:
How about subscribing a function (not anonymous one) - so that in a handler we can deregister ourselves?
That way we'd avoid double invocations and would be able to get rid of the Disposing logic